src/components/_classes/component/Component.js

```diff
@@ -2279,7 +2279,9 @@ export default class Component extends Element {
     }
     this.calculateComponentValue(data, flags, row);
     this.checkComponentConditions(data, flags, row);
-    return flags.noValidate ? true : this.checkComponentValidity(data, false, row);
+    const shouldCheckValidity = !this.builderMode && !this.options.preview && this.defaultValue;
+    const isValid =  shouldCheckValidity ? this.checkComponentValidity(data) : true;
```

Comment:
Why are we performing two ```checkComponentValidity``` checks now?

Reply:
What do you mean?
I have only one ```checkComponentValidity``` in line 2283.

Reply:
Did some performance improvement.